### PR TITLE
Update dependencies (1.14.3 edition)

### PR DESCRIPTION
**Pull request by Khitiara**
Also fixed a few bugs

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
+yarn_mappings=1.14.3+build.7
+loader_version=0.4.8+build.154
+# Mod Properties
+mod_version=0.3.0
+maven_group=net.fabricmc
+archives_base_name=fabric-language-scala
+description=Fabric language module for Scala
+url=https://github.com/FabricMC/fabric-language-scala/
+# Other
+scala_version=2.12.8
```
Comment (modmuss50) [inline gradle.properties]: Might as well bump this to 2.13.0?

File: build.gradle [hunk 2]
```diff
@@ -1,125 +1,111 @@
...
-	id "idea"
-	id "maven"
-	id "maven-publish"
-	id "com.github.johnrengelman.shadow" version "2.0.4"
-	id 'com.matthewprenger.cursegradle' version "1.1.2"
+    id "java"
+    id "scala"
+    id "maven"
+    id "maven-publish"
+    id 'fabric-loom' version '0.2.2-SNAPSHOT'
```
Comment (modmuss50) [inline build.gradle]: Idealy this should be using 0.2.3 or higher as that has the improved dependencies in the maven pom.

**Comment by modmuss50** [comment]
Im getting an error when trying to build this: 

```
FAILURE: Build failed with an exception.

* Where:
Build file 'C:\Users\mark\Documents\fabric-language-scala\build.gradle' line: 62

* What went wrong:
A problem occurred evaluating root project 'fabric-language-scala'.
> Could not set unknown property 'jar' for task ':remapJar' of type net.fabricmc.loom.task.RemapJarTask.
```

**Comment by robotbrain** [comment]
That error is why I was still on Loom 0.2.2 as i recall

**Comment by modmuss50** [comment]
Humm ok, I can look into updating it if you want, as loom 0.2.2 is really old and shouldnt be used.

**Comment by robotbrain** [comment]
If I knew how to get the remapJar task to use a different input jar itd be an easy fix, i can do it if youd prefer if I can figure that out

**Comment by modmuss50** [comment]
ive done it, 0.2.5 rewrote it to make a it lot easier so ive done it with that like so:

```groovy
remapJar {
	dependsOn tasks.shadowJar
	input = shadowJar.archivePath
}
```

**Comment by robotbrain** [comment]
Alright, works for me then. Pushed the same fix from my end

**Comment by modmuss50** [comment]
Seems like there is a gradle issue preventing 0.13.0 from building as well

[https://github.com/gradle/gradle/issues/9857](https://github.com/gradle/gradle/issues/9857)

Guess I will roll that back as well, I will finish off the PR and merge it in, thanks.

**Comment by robotbrain** [comment]
Alright, and thank you ;)

**Comment by modmuss50** [comment]
This needs a fair amount more work than I thought tbh,

The none `fat` jar contains all the deps.
The `fat` jar was renamed to `all`
The maven pom does not contain the scala libs.

I will see about fixing these before merging, I may not be able to do it tonight. Thanks for the help anyway.

**Comment by modmuss50** [comment]
Most of the changes from this were already present, it is now quite outdated. Closing the PR, thanks.

**Pull request closed without merging**